### DISCUSSION TITLE
Line Numbering Code Chunks in MS Word (docx) Documents?

### DISCUSSION THREAD

**Discussion by mlamias** [quarto-dev/quarto-cli#4362] [Q&A]
I was trying to create a Word document in Quarto today to display some code chunks with syntax highlighting and line numbering, but I realized after reading the documentation and after seeing no line numbers appear in my document that this is not supported for MS Word.  Are there any plans to build out this feature?  If so, is there any ETA yet on when this will be implemented?  I only ask because I'm about to embark on a large project that really needs to have the line numbering feature in Word, so I may have to look for another method to accomplish this, but if this will be released soon, I'll hold off on development for this feature.

Thank you!

**Comment by mcanouil** [comment]
Well, it's probably "easier" for you to create the Word template you need then give it to Quarto/Pandoc.
See <https://quarto.org/docs/output-formats/ms-word-templates.html>.

The thing is that Pandoc uses a Word template and does not really create it, but instead fill one.
Since Quarto relies on Pandoc, Word format will likely be always behind format like LaTeX/pdf and HTML.

**Reply by dragonstyle** [reply]
Additionally, I opened an enhancement request for Pandoc to support this here:

https://github.com/jgm/pandoc/issues/8627

No ETA and we'd really want Pandoc to support this directly, so perhaps best to plan an alternative, unfortunately.

**Reply by mlamias** [reply]
Sadly, I've tried the template approach and can't get it to work the way I want to.  It seems the word templates will only apply line numbering styles to an entire section on page.  I can't seem to get and Template Style to **only** number code blocks, and re-number the line numbers with each subsequent code block.

**Reply by mcanouil** [reply]
Maybe a Lua filter but for this kind of stuff, it would be really tricky to code it to work as expected.
